src/openms/include/OpenMS/CHEMISTRY/NASequence.h

```diff
@@ -0,0 +1,504 @@
+      typedef std::vector<const value_type*>::difference_type difference_type;
+
+      /** @name Constructors and destructors
+       */
+      //@{
+      Iterator() = default;
+
+      /// detailed constructor with pointer to the vector and offset position
+      Iterator(std::vector<const Ribonucleotide*>* vec_ptr,
+               difference_type position)
+      {
+        vector_ = vec_ptr;
+        position_ = position;
+      }
+
+      /// copy constructor
+      Iterator(const Iterator& rhs) :
+        vector_(rhs.vector_),
+        position_(rhs.position_)
+      {
+      }
+
+      /// destructor
+      virtual ~Iterator() {}
+
+      //@}
+
+      /// assignment operator
+      Iterator& operator=(const Iterator& rhs)
+      {
+        if (this != &rhs)
+        {
+          position_ = rhs.position_;
+          vector_ = rhs.vector_;
+        }
+        return *this;
+      }
+
+      /** @name Operators
+       */
+      //@{
+      /// dereference operator
+      const_reference operator*() const
+      {
+        return *(*vector_)[position_];
+      }
+
+      /// dereference operator
+      const_pointer operator->() const
+      {
+        return (*vector_)[position_];
+      }
+
+      /// mutable dereference operator
+      pointer operator->()
+      {
+        return (*vector_)[position_];
+      }
+
+      /// forward jump operator
+      const Iterator operator+(difference_type diff) const
+      {
+        return Iterator(vector_, position_ + diff);
+      }
+
+      difference_type operator-(Iterator rhs) const
+      {
+        return position_ - rhs.position_;
+      }
+
+      /// backward jump operator
+      const Iterator operator-(difference_type diff) const
+      {
+        return Iterator(vector_, position_ - diff);
+      }
+
+      /// equality comparator
+      bool operator==(const Iterator& rhs) const
+      {
+        return (std::tie(vector_,position_) ==
+                std::tie(rhs.vector_, rhs.position_));
+      }
+
+      /// inequality operator
+      bool operator!=(const Iterator& rhs) const
+      {
+        return !this->operator==(rhs);
+      }
+
+      /// increment operator
+      Iterator& operator++()
+      {
+        ++position_;
+        return *this;
+      }
+
+      /// decrement operator
+      Iterator& operator--()
+      {
+        --position_;
+        return *this;
+      }
+
+      //@}
+
+    protected:
+
+      std::vector<const Ribonucleotide*>* vector_;
+
+      // position in the vector
+      difference_type position_;
+    };
+
+  public:
+    /*
+     * Default constructors and assignment operators.
+     */
+    NASequence() = default; /// default constructor
+    NASequence(const NASequence&) = default; // Copy constructor
+    NASequence(NASequence&&) = default; // Move constructor
+    NASequence& operator=(const NASequence&) & = default;  // Copy assignment operator
+    NASequence& operator=(NASequence&&) & = default; // Move assignment operator
```

Comment:
please use `///<` for this, right now doxygen will annotate `NASequence(const NASequence&) = default;` with "default constructor"